bson/bsonrw/extjson_writer.go

```diff
@@ -536,7 +536,10 @@ func (ejvw *extJSONValueWriter) WriteUndefined() error {
 func (ejvw *extJSONValueWriter) WriteDocumentElement(key string) (ValueWriter, error) {
 	switch ejvw.stack[ejvw.frame].mode {
 	case mDocument, mTopLevel, mCodeWithScope:
-		ejvw.buf = append(ejvw.buf, []byte(fmt.Sprintf(`"%s":`, key))...)
+		var buf bytes.Buffer
+		writeStringWithEscapes(key, &buf, ejvw.escapeHTML)
```

Comment:
Will this also ensure that keys like `"\usb#"` will be escaped? I ran into a problem with that specifically and then when I saw this bug I came up with the simpler test case you made above.

I figure that should be written as `"\\usb#"`? Is it worth adding a test case to make sure this works as well?

Reply:
I just changed the test case to the more complex one.

Reply:
cool!